liblwgeom/gbox.c

```diff
@@ -778,15 +851,20 @@ uint64_t gbox_get_sortable_hash(const GBOX *g)
 	}
 	else
 	{
-		/*
-		* Here we'd like to get two ordinates from 4 in the box.
-		* Since it's just a sortable bit representation we can omit division from (A+B)/2.
-		* All it should do is subtract 1 from exponent anyways.
-		*/
-		x.f = g->xmax + g->xmin;
-		y.f = g->ymax + g->ymin;
+		x.f = (g->xmax + g->xmin) / 2;
+		y.f = (g->ymax + g->ymin) / 2;
 	}
-	return uint32_interleave_2(x.u, y.u);
-}
 
+	if (srid == 3857)
```

Comment:
Can you please leave a comment explaining what this does and why it's needed?

Reply:
Yeah, this is pretty magical. It also suffers from a problem which I also have in the tile envelope in that it assumes an SRID number equals a particular CRS, which it does not, it equals a particular row in the `spatial_ref_sys` table. The SRID number needs to be looked up against `spatial_ref_sys` and then checked to see if the auth/auth_srid combo is the one you expect. At a minimum, put a big "to do" comment there, and I will do the same. I think w/ the right use of the existing projection caching mechanism it shouldn't be too terribly bad.

Reply:
Added a comment.
For the SRID, if user redefines the row with one of SRIDs from here it doesn't break much - just the curve will wind a bit differently. Someone pushing lonlat into 3857 will see a hilbert-sorted precision-limit boxes sorted by x,y, and someone pushing mercator into 4326 will see the close-to-0 artifacts in a bit different spot, kilometer away, scaled.
Benefit is much higher: no artifacts if srid assigned in expected way :)